pkg/collector/corechecks/cluster/topologycollectors/persistent_volume_source_mapper.go

```diff
@@ -0,0 +1,398 @@
+package topologycollectors
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/StackVista/stackstate-agent/pkg/topology"
+	v1 "k8s.io/api/core/v1"
+)
+
+// PersistentVolumeSourceMapper maps a PersistentVolumeSource to an external Volume topology component
+type PersistentVolumeSourceMapper func(pc *PersistentVolumeCollector, volume v1.PersistentVolume) (*topology.Component, error)
+
+var allPersistentVolumeSourceMappers = []PersistentVolumeSourceMapper{
+	mapAwsEbsPersistentVolume,
+	mapAzureDiskPersistentVolume,
+	mapAzureFilePersistentVolume,
+	mapCephFsPersistentVolume,
+	mapCinderPersistentVolume,
+	mapFCPersistentVolume,
+	mapFlexPersistentVolume,
+	mapFlockerPersistentVolume,
+	mapGcePersistentDiskPersistentVolume,
+	mapGlusterFsPersistentVolume,
+	mapIscsiPersistentVolume,
+	mapNfsPersistentVolume,
+	mapPhotonPersistentDiskPersistentVolume,
+	mapPortWorxPersistentVolume,
+	mapQuobytePersistentVolume,
+	mapRbdPersistentVolume,
+	mapScaleIoPersistentVolume,
+	mapStorageOsPersistentVolume,
+	mapVspherePersistentVolume,
+}
+
+func mapAwsEbsPersistentVolume(pc *PersistentVolumeCollector, volume v1.PersistentVolume) (*topology.Component, error) {
+	if volume.Spec.AWSElasticBlockStore == nil {
+		return nil, nil
+	}
+
+	extID := pc.GetURNBuilder().BuildExternalVolumeExternalID("aws-ebs", strings.TrimPrefix(volume.Spec.AWSElasticBlockStore.VolumeID, "aws://"), fmt.Sprint(volume.Spec.AWSElasticBlockStore.Partition))
+
+	tags := map[string]string{
+		"kind":      "aws-ebs",
+		"volume-id": volume.Spec.AWSElasticBlockStore.VolumeID,
```

Comment:
Too bad we don't (and neither does AWS have an arn) have a standardized identifier yet for AWS EBS volumes. Otherwise we could include that here as an identifier already and have it merge whenever it gets added to the AWS Stackpack.